Streetcode/Streetcode.BLL/MediatR/Streetcode/Streetcode/Update/UpdateStreetcodeHandler.cs

```diff
@@ -230,6 +237,111 @@ private void UpdateAudio(IEnumerable<AudioUpdateDTO> audios, StreetcodeContent s
             }
         }
 
+        private async Task UpdateArtGallery(StreetcodeContent streetcode, IEnumerable<StreetcodeArtSlideCreateUpdateDTO> artSlides, IEnumerable<ArtCreateUpdateDTO> arts)
+        {
+            _repositoryWrapper.StreetcodeArtRepository.DeleteRange(await _repositoryWrapper.StreetcodeArtRepository.GetAllAsync(a => a.StreetcodeId == streetcode.Id));
+
+            var toCreateArts = new List<Art>();
+            var toDeleteArts = new List<Art>();
+            var toUpdateArts = new List<Art>();
+            var oldArtIds = new List<int>();
+            foreach (var art in arts)
+            {
+                var newArt = new Art
+                {
+                    Description = art.Description,
+                    ImageId = art.ImageId,
+                    StreetcodeId = streetcode.Id,
+                    Title = art.Title
+                };
+                if (art.ModelState == ModelState.Created)
+                {
+                    oldArtIds.Add(art.Id);
+                    toCreateArts.Add(newArt);
+                }
+                else
+                {
+                    newArt.Id = art.Id;
+                    if (art.ModelState == ModelState.Deleted)
+                    {
+                        toDeleteArts.Add(newArt);
+                    }
+                    else
+                    {
+                        toUpdateArts.Add(newArt);
+                    }
+                }
+            }
+
+            await _repositoryWrapper.ArtRepository.CreateRangeAsync(toCreateArts);
+            _repositoryWrapper.ArtRepository.DeleteRange(toDeleteArts);
+            _repositoryWrapper.ArtRepository.UpdateRange(toUpdateArts);
+            await _repositoryWrapper.SaveChangesAsync();
+
+            var toCreateSlides = new List<StreetcodeArtSlide>();
+            var toDeleteSlides = new List<StreetcodeArtSlide>();
```

Comment:
Too hard to read and understand this part of code from line 280 to 337 (foreach loop). Try to break this code to separate private functions with proper names to make this code better